source/SkiaSharp.Views.Uno/SkiaSharp.Views.Uno.Wasm/WasmScripts/SkiaSharp.Views.Uno.Wasm.js

```diff
@@ -22,7 +22,109 @@
                     return true;
                 }
             }
+
+            class SKSwapChainPanel {
+                static activeInstances = {};
+
+                constructor(managedHandle) {
+                    this.managedHandle = managedHandle;
+                    this.canvas = undefined;
+                    this.jsInfo = undefined;
+                }
+
+                // JSObject
+                static createInstance(managedHandle, jsHandle) {
+                    SKSwapChainPanel.activeInstances[jsHandle] = new SKSwapChainPanel(managedHandle);
+                }
+                static getInstance(jsHandle) {
+                    return SKSwapChainPanel.activeInstances[jsHandle];
+                }
+                static destroyInstance(jsHandle) {
+                    delete SKSwapChainPanel.activeInstances[jsHandle];
+                }
+
+                requestAnimationFrame() {
+                    if (this.canvas) {
+                        var scale = window.devicePixelRatio || 1;
+                        this.canvas.width = this.canvas.clientWidth * scale;
+                        this.canvas.height = this.canvas.clientHeight * scale;
+                    }
+
+                    window.requestAnimationFrame(() => {
+                        Uno.Foundation.Interop.ManagedObject.dispatch(this.managedHandle, 'RenderFrame', null);
```

Comment:
I think this is correct to call back into an Uno `JSObject`.

Reply:
@dr1rrb Looks good to you ?

Reply:
Yeah it's the good way 👍 (I would love to generate this kind of glue between C# and Javascript, but didn't take time to do it yet)